tables/elb_access_log/elb_access_log.go

```diff
@@ -0,0 +1,214 @@
+package elb_access_log
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/turbot/tailpipe-plugin-sdk/schema"
+)
+
+type ElbAccessLogBatch struct {
+	Records []ElbAccessLog `json:"Records"`
+}
+
+type ElbAccessLog struct {
+	schema.CommonFields
+
+	ActionsExecuted        string    `json:"actions_executed"`
+	ChosenCertArn          string    `json:"chosen_cert_arn"`
+	Classification         *string   `json:"classification,omitempty"`
+	ClassificationReason   *string   `json:"classification_reason,omitempty"`
+	ClientIP               string    `json:"client_ip"`
+	ClientPort             int       `json:"client_port"`
+	ConnTraceID            *string   `json:"conn_trace_id,omitempty"`
+	DomainName             string    `json:"domain_name"`
+	Elb                    string    `json:"elb"`
+	ElbStatusCode          *int      `json:"elb_status_code,omitempty"`
+	ErrorReason            *string   `json:"error_reason,omitempty"`
+	MatchedRulePriority    int       `json:"matched_rule_priority"`
+	ReceivedBytes          *int64    `json:"received_bytes"`
+	RedirectURL            *string   `json:"redirect_url,omitempty"`
+	Request                string    `json:"request"`
+	RequestCreationTime    time.Time `json:"request_creation_time"`
+	RequestProcessingTime  float64   `json:"request_processing_time"`
+	ResponseProcessingTime float64   `json:"response_processing_time"`
+	SentBytes              *int64    `json:"sent_bytes"`
+	SslCipher              string    `json:"ssl_cipher"`
+	SslProtocol            string    `json:"ssl_protocol"`
+	TargetGroupArn         string    `json:"target_group_arn"`
+	TargetIP               *string   `json:"target_ip,omitempty"`
+	TargetList             *string   `json:"target_list,omitempty"`
+	TargetPort             int       `json:"target_port,omitempty"`
+	TargetProcessingTime   float64   `json:"target_processing_time"`
+	TargetStatusCode       *int      `json:"target_status_code,omitempty"`
+	TargetStatusList       *string   `json:"target_status_list,omitempty"`
+	Timestamp              time.Time `json:"timestamp"`
+	TraceID                string    `json:"trace_id"`
+	Type                   string    `json:"type"`
+	UserAgent              string    `json:"user_agent"`
+}
+
+// InitialiseFromMap - initialise the struct from a map
+func (l *ElbAccessLog) InitialiseFromMap(m map[string]string) error {
+	var err error
+	for key, value := range m {
+		if value == "-" {
```

Comment:
I like how we do the check here, can we also do something similar in the S3 access log table?